ykrt/src/compile/jitc_yk/jit_ir/mod.rs

```diff
@@ -106,6 +106,11 @@ impl Module {
         Self::new_internal(ctr_id, global_decls_len)
     }
 
+    #[cfg(any(debug_assertions, test))]
+    pub(crate) fn id(&self) -> u64 {
```

Comment:
Remember that this ID is not guaranteed to be stable: I think it's worth documenting that (again) on this function (see the doc string comment on the attribute).

Reply:
Must admit I missed that detail.

Would it be hard to make this an overflowing sequential integer? `gdb_c_test` assumes this for this feature:

```
  -b, --num-breaks <NUM_BREAKS>  Set breakpoints at the first `N` compiled traces
```

(this assumes I can get break points on JIT code working. It doesn't seem to work right now).

Reply:
I think the current guarantees are the correct ones, and the gdb backend should just accept that things might go a bit wonky if you run things for a billion years or so.

Reply:
[You may notice from my mild attempt at humour that I don't think this is an issue in practise for the gdb backend, but the `id` function should document the guarantee -- or lack thereof -- anyway.]

Reply:
> I guess that's ok, since we'd never gdb a rust test?

Hmm, we might want to do that at some point, but far into the future. Maybe just make `#[cfg(test)]` not compile in the GDB code so that we stop our future selves being confused? Or at least `FIXME` or similar perhaps?

Reply:
We could get the gdb plugin to generate its own sequential numbers, then the problem is uniform for all modes of operation. Since gdb is single-threaded (right?) I don't think we'd even have to synchronise updates to the counter.

(If we did that though, I'm unsure what role `ctr_id` plays any more)

Reply:
We're over-thinking this IMHO: I just think the `id` function in `jit_ir`, which isn't gdb specific, should document what it does. The gdb backend is, in practise, *fine* with the guarantee for the foreseeable future: I just don't want the `id` function to be misused by someone else in the medium term.